src/Build/BackEnd/Components/RequestBuilder/TaskBuilder.cs

```diff
@@ -337,7 +341,12 @@ private async Task<WorkUnitResult> ExecuteTask(TaskExecutionMode mode, Lookup lo
                         break;
                     }
                 }
-
+                // Some tests do not provide an actual taskNode; checking if _taskNode == null prevents those tests from failing.
+                if (MSBuildEventSource.Log.IsEnabled())
+                {
+                    long endTime = MSBuildEventSource.GetThreadTime();
```

Comment:
@Forgind mentioned that some tests were coming up with negative times in perfview. @brianrob would you happen to know any reason why this would happen?

Inspiration for this idea came from [here](https://www.codeproject.com/Articles/31152/ExecutionStopwatch)

Reply:
I experimented some more, and with the latest commit, they now show up as always non-negative (as far as I can tell). I think the problem was that, as I noted in our call, there can be overflow between if a "stopwatch" starts near the max time and ends shortly after it, since that would seem like "negative" time. When I used TimeSpan.Duration(), that problem went away.

Reply:
I would strongly recommend that rather than calculating times in your code that you do this calculation on the resulting trace.  If you're using PerfView it will do the calculation for you automatically in the Events view.

Reply:
That's DURATION_MSEC, I imagine? I'd thought that was wall clock time rather than execution time. @danmosemsft, @cdmihai, does that resolve the IBuildEngineX.Yield problem?

Reply:
Are you trying to capture CPU time?  If so, I'd recommend using an ETW trace that captures CPU samples rather than asking the kernel for this information.  The reason is that what ends up happening is you get a trace that contains the numbers, but then the next question is "where is that time spent?", and you can answer that with a CPU trace, and the barrier to collection of CPU events is the same as collecting the new event that you're adding here.